From: =?UTF-8?q?Mika=20Pfl=C3=BCger?= <debian@mikapflueger.de>
Date: Fri, 2 Mar 2012 02:07:00 +0100
Subject: mozilla.{fc,te}: fixes for debian mozilla/iceweasel packaging and
 fix to use userdom_search_user_home_conent instead of
 userdom_serach_user_home_dirs.

---
 policy/modules/apps/mozilla.fc |    4 ++++
 policy/modules/apps/mozilla.te |    4 +++-
 2 files changed, 7 insertions(+), 1 deletions(-)

Index: refpolicy-2.20110726/policy/modules/apps/mozilla.fc
===================================================================
--- refpolicy-2.20110726.orig/policy/modules/apps/mozilla.fc	2012-06-21 15:30:37.000000000 +1000
+++ refpolicy-2.20110726/policy/modules/apps/mozilla.fc	2012-06-21 15:30:37.000000000 +1000
@@ -20,6 +20,9 @@
 /usr/bin/epiphany		--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/bin/mozilla-[0-9].*	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/bin/mozilla-bin-[0-9].*	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+ifdef(`distro_debian', `
+/usr/lib/iceweasel/iceweasel	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+')
 
 #
 # /lib
@@ -35,3 +38,4 @@
 /usr/lib64/[^/]*firefox[^/]*/firefox -- gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/lib/chromium(-browser)?/chromium(-browser)?-sandbox -- gen_context(system_u:object_r:chrome_sandbox_exec_t,s0)
 /usr/lib/chromium(-browser)?/chromium(-browser)? -- gen_context(system_u:object_r:chrome_browser_exec_t,s0)
+/usr/lib/xulrunner-1.9.1/xulrunner-stub -- gen_context(system_u:object_r:mozilla_exec_t,s0)
Index: refpolicy-2.20110726/policy/modules/apps/mozilla.te
===================================================================
--- refpolicy-2.20110726.orig/policy/modules/apps/mozilla.te	2012-06-21 15:30:37.000000000 +1000
+++ refpolicy-2.20110726/policy/modules/apps/mozilla.te	2012-06-21 15:31:54.818072677 +1000
@@ -110,6 +110,7 @@
 dev_read_sysfs(mozilla_t)
 domain_dontaudit_search_all_domains_state(chrome_sandbox_t)
 
+files_dontaudit_search_boot(mozilla_t)
 allow mozilla_t self:capability { sys_nice setgid setuid };
 allow mozilla_t self:process { sigkill signal setsched getsched setrlimit };
 allow mozilla_t self:fifo_file rw_fifo_file_perms;
@@ -128,7 +129,7 @@
 manage_dirs_pattern(mozilla_t, mozilla_home_t, mozilla_home_t)
 manage_files_pattern(mozilla_t, mozilla_home_t, mozilla_home_t)
 manage_lnk_files_pattern(mozilla_t, mozilla_home_t, mozilla_home_t)
-userdom_search_user_home_dirs(mozilla_t)
+userdom_search_user_home_content(mozilla_t)
 userdom_user_home_dir_filetrans(mozilla_t, mozilla_home_t, dir)
 
 # Mozpluggerrc
@@ -136,6 +137,8 @@
 
 manage_files_pattern(mozilla_t, mozilla_tmp_t, mozilla_tmp_t)
 manage_dirs_pattern(mozilla_t, mozilla_tmp_t, mozilla_tmp_t)
+manage_lnk_files_pattern(mozilla_t, mozilla_tmp_t, mozilla_tmp_t)
+manage_sock_files_pattern(mozilla_t, mozilla_tmp_t, mozilla_tmp_t)
 files_tmp_filetrans(mozilla_t, mozilla_tmp_t, { file dir })
 
 manage_files_pattern(mozilla_t, mozilla_tmpfs_t, mozilla_tmpfs_t)
